dashboard/controller/assignments/addAssignment.js

```diff
@@ -0,0 +1,121 @@
+        req.body.name = req.body.name.trim();
+        req.body.classrooms = req.body.classrooms || [];
+        if (typeof req.body.classrooms == 'string') {
+            req.body.classrooms = [req.body.classrooms];
+        }
+        // req.body.color = JSON.parse(req.body.color);
+        req.assert('name', common.l10n.get('UsernameInvalid')).matches(/^[a-z0-9 ]+$/i);
+        req.body.options = { sync: true, stats: true };
+
+        // get errors
+        var errors = req.validationErrors();
+
+        // call
+        if (!errors) {
+            superagent
+                .post(common.getAPIUrl(req) + 'api/v1/assignments')
+                .set(common.getHeaders(req))
+                .send({
+                    assignment: JSON.stringify(req.body)
+                })
+                .end(function (error, response) {
+                    if (response.statusCode == 200) {
+                        console.log({ "body": response.body });
+                        // send to classrooms page
+                        req.flash('success', {
+                            msg: common.l10n.get('AssignmentCreated', { name: req.body.name })
+                        });
+                        return res.redirect('/dashboard/assignments/');
+                    } else {
+                        req.flash('errors', {
+                            msg: common.l10n.get('ErrorCode' + response.body.code)
+                        });
+                        return res.redirect('/dashboard/assignments/add');
+                    }
+                });
+        } else {
+            dashboard_utils.getAllClassrooms(req, res, function (classrooms) {
+                journal_utils.getJournalEntries(req, res, query, function (journalEntries) {
+                    console.log({ "journalEntries": journalEntries });
+                    if (req.body.classrooms && typeof (req.body.classrooms) == "object" && req.body.classrooms.length > 0 && classrooms && classrooms.classrooms && classrooms.classrooms.length > 0) {
+                        for (var i = 0; i < classrooms.classrooms.length; i++) {
+                            if (req.body.classrooms.indexOf(classrooms.classrooms[i]._id) != -1) {
+                                classrooms.classrooms[i]['is_member'] = true;
+                            }
+                        }
+                    }
+                    req.flash('errors', errors);
+                    return res.render('addEditAssignment', {
+                        mode: "add",
+                        module: 'assignments',
+                        xocolors: xocolors,
+                        emoji: emoji,
+                        moment: moment,
+                        entries: journalEntries.entries,
+                        classrooms: classrooms.classrooms,
+                        account: req.session.user,
+                        server: assignment.ini().information
+                    });
+                });
+            });
```

Comment:
Have you manually verified that on submitting the form with errors, the page reloads without any loss in the information already filled in the form?

Reply:
No, the information lost after page reload with error message !

Reply:
But the other forms retain the form information, right? We need to forms features to be as similar as possible.

Reply:
Done